.github/workflows/check.yml

```diff
@@ -0,0 +1,41 @@
+# Based upon:
+#   https://github.com/actions/starter-workflows/blob/main/ci/go.yml
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.18, 1.19]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Download Packages
+        run: go get -t -v
```

Comment:
you don't need this step anymore.

Reply:
Why not?

Reply:
because since Go 1.11, `go build` downloads all the dependancies needed. 
So you no longer need a `go get` step in the build.

Reply:
White that's true, a separate step in CI makes it more clear at which step the error occurred.